### PR TITLE
Adding Java 11 recipe -RemovedSecurityManagerMethods

### PR DESCRIPTION
**Pull request by ranuradh**
## What's changed?
Added recipe for the following:
image

This recipe replaces methods returning `void` that are deprecated in the `SecurityManager` class in Java SE 11. 
The methods `checkAwtEventQueueAccess()`,`checkSystemClipboardAccess()`,`checkMemberAccess()` and `checkTopLevelWindow()` are replaced by `checkPermission(new java.security.AllPermission())`.

Since it was not possible to test the recipe. I created sample files, created a local plugin using 
` ./gradlew publishToMavenLocal` then ran it in the Java8 Sample App using `mvn rewrite:dryRun`

Attaching the rewrite.patch file
[rewrite.patch](https://github.com/openrewrite/rewrite-migrate-java/files/15028204/rewrite.patch)

### Checklist
- [X] I've added unit tests to cover both positive and negative cases
- [X] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [X] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: src/main/java/org/openrewrite/java/migrate/RemovedSecurityManagerMethods.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+            private final MethodMatcher METHOD_PATTERN_CLIP = new MethodMatcher("java.lang.SecurityManager checkSystemClipboardAccess()", false);
+            private final MethodMatcher METHOD_PATTERN_MEMBER = new MethodMatcher("java.lang.SecurityManager checkMemberAccess(..)", false);
+            private final MethodMatcher METHOD_PATTERN_WINDOW = new MethodMatcher("java.lang.SecurityManager checkTopLevelWindow(..)", false);
+            @Override
+            public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
+                if ((METHOD_PATTERN_QUE.matches(method) || (METHOD_PATTERN_CLIP).matches(method) || (METHOD_PATTERN_MEMBER).matches(method)) || (METHOD_PATTERN_WINDOW).matches(method)) {
+                    return JavaTemplate.builder("checkPermission(new java.security.AllPermission())")
+                            .imports("java.security.AllPermission")
+                            .build().apply(updateCursor(method),
+                                    method.getCoordinates().replaceMethod());
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/migrate/RemovedSecurityManagerMethods.java:50]: Why did you add the additional braces when every operator is a logical or?
```suggestion
            private final MethodMatcher METHOD_PATTERN_QUE = new MethodMatcher("java.lang.SecurityManager checkAwtEventQueueAccess()", false);
            private final MethodMatcher METHOD_PATTERN_CLIP = new MethodMatcher("java.lang.SecurityManager checkSystemClipboardAccess()", false);
            private final MethodMatcher METHOD_PATTERN_MEMBER = new MethodMatcher("java.lang.SecurityManager checkMemberAccess(..)", false);
            private final MethodMatcher METHOD_PATTERN_WINDOW = new MethodMatcher("java.lang.SecurityManager checkTopLevelWindow(..)", false);

                        @Override
            public J visitMethodInvocation(J.MethodInvocation method, ExecutionContext ctx) {
                if (METHOD_PATTERN_QUE.matches(method) || METHOD_PATTERN_CLIP.matches(method) || METHOD_PATTERN_MEMBER.matches(method) || METHOD_PATTERN_WINDOW.matches(method)) {
                    return JavaTemplate.builder("checkPermission(new java.security.AllPermission())")
                            .imports("java.security.AllPermission")
                            .build().apply(updateCursor(method), method.getCoordinates().replaceMethod());
```

Comment (timtebeek) [reply src/main/java/org/openrewrite/java/migrate/RemovedSecurityManagerMethods.java:50]: I'm also surprised at not seeing a unit test; where there any complications with writing one?
In particular the qualified use of `java.security.AllPermission` we might want to replace with an import.

File: src/main/java/org/openrewrite/java/migrate/RemovedSecurityManagerMethods.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+public class RemovedSecurityManagerMethods extends Recipe {
+    @Override
+    public String getDisplayName() {
+        return "Replace deprecated methods in`SecurityManager`";
+    }
+    @Override
+    public String getDescription() {
+        return "This recipe replaces methods returning `void` that are deprecated in the SecurityManager class in Java SE 11." +
+                "The methods `checkAwtEventQueueAccess()`,`checkSystemClipboardAccess()`,`checkMemberAccess()` and `checkTopLevelWindow()`" +
+                " are replaced by `checkPermission(new java.security.AllPermission())`.";
```
Comment (timtebeek) [inline src/main/java/org/openrewrite/java/migrate/RemovedSecurityManagerMethods.java:35]:
```suggestion
        return "Replace `SecurityManager` methods `checkAwtEventQueueAccess()`, `checkSystemClipboardAccess()`, `checkMemberAccess()` and `checkTopLevelWindow()` deprecated in Java SE 11 by `checkPermission(new java.security.AllPermission())`.";
```

File: src/main/java/org/openrewrite/java/migrate/RemovedSecurityManagerMethods.java (hunk 1 above)
Comment (ranuradh) [reply src/main/java/org/openrewrite/java/migrate/RemovedSecurityManagerMethods.java]: Sorry fixed the additional braces in the next commit..was probably some copy paste remanent.  
The issue with writing the unit tests for this one was since it was a jdk change, I was not able to include the .jar (jdk7-2.1.7.jar) in the tests to compile using the older jar. We found a similar issue in the past having older version j2se tests. Hence I came up with this method.

File: src/main/java/org/openrewrite/java/migrate/RemovedSecurityManagerMethods.java (hunk 2 above)
Comment (ranuradh) [reply src/main/java/org/openrewrite/java/migrate/RemovedSecurityManagerMethods.java]: Fixed this in the next commit

**Comment by ranuradh** [comment]
@timtebeek and @cjobinabo could we complete the review and merge this. Since I believe we need this for our next release.

**Review by timtebeek — APPROVED** [review]
Thanks a lot, also for the context around the challenges to test this one. Ran a local test and it all works well.
